addons/promote/common/src/main/data/promote/rules/maven-no-pre-existing-paths.groovy

```diff
@@ -12,12 +14,13 @@ class MavenNoPreExistingPaths implements ValidationRule {
 
         def errors = Collections.synchronizedList(new ArrayList());
         def tools = request.getTools()
-
         tools.paralleledEach(request.getSourcePaths(), { it ->
             def aref = tools.getArtifact(it);
             if (aref != null) {
                 tools.forEach(verifyStoreKeys, { verifyStoreKey ->
-                    if (tools.exists(verifyStoreKey, it)) {
+                    if (tools.exists(verifyStoreKey, it)
+                            && !(tools.digest(verifyStoreKey, it, MavenPackageTypeDescriptor.MAVEN_PKG_KEY).get(ContentDigest.SHA_256)
+                            .equals(tools.digest(request.getPromoteRequest().getSource(), it, MavenPackageTypeDescriptor.MAVEN_PKG_KEY).get(ContentDigest.SHA_256)))) {
                         errors.add(String.format("%s is already available in: %s", it, verifyStoreKey))
```

Comment:
Please change to "... is already available with different checksum in..." to make it clear what's the issue.

Reply:
Yeah, that makes sense.